### PR TITLE
CANDLEPIN-292: Introduce Centos Stream 9 vagrant setup

### PR DESCRIPTION
**Pull request by Januson**
- Add EL9 support
- Drop EL7 support
- Drop Fedora 35 support

### CODE REVIEW

File: ansible/roles/candlepin/tasks/main.yml [hunk 1]
```diff
@@ -5,32 +5,33 @@
...
+    file: el8/el8_tasks.yml
   when:
-    - ansible_facts['distribution'] in ["CentOS", "RedHat"]
-    - ansible_facts['distribution_major_version'] == "7"
+    - ansible_distribution in ["CentOS", "RedHat"]
+    - ansible_distribution_major_version == "8"
   tags:
     - always
 
 - name: "Include EL8 tasks"
```
Comment (Ceiu) [inline ansible/roles/candlepin/tasks/main.yml:17]: EL9 task still labeled as EL8

File: ansible/roles/candlepin/tasks/common/cp_mariadb.yml [hunk 2]
```diff
@@ -88,6 +91,20 @@
     name: candlepin
     priv: "candlepin.*:ALL,GRANT"
     state: present
+  when:
```
Comment (Ceiu) [inline ansible/roles/candlepin/tasks/common/cp_mariadb.yml:94]: If we're stuck doing part of these DB changes differently based on the OS, I think I'd rather leave the whole thing in each directory as it was -- especially if we're going to be staying at only 2 or 3 different machines.

**Review by Ceiu — CHANGES_REQUESTED** [review]
Couple very minor changes, but otherwise looks good.

Unfortunately, my hope for the DB changes was not as clean with the removal of the other VMs (specifically EL7) -- I was hoping that would just be identical going forward. But if we're stuck having to still look at the underlying OS and branch behavior, we're still better off in terms of readability leaving the tasks in their OS-specific variants.

File: ansible/roles/candlepin/tasks/main.yml (hunk 1 above)
Comment (Januson) [reply ansible/roles/candlepin/tasks/main.yml]: Done.

File: ansible/roles/candlepin/tasks/common/cp_mariadb.yml (hunk 2 above)
Comment (Januson) [reply ansible/roles/candlepin/tasks/common/cp_mariadb.yml]: Done.

**Review by Ceiu — APPROVED** [review]

**Pull request merged**